tracer/test/Datadog.Trace.Tests/ManualInstrumentation/SettingsInstrumentationTests.cs

```diff
@@ -0,0 +1,305 @@
+        manual.SetHttpClientErrorStatusCodes([400, 401, 402]);
+        manual.SetHttpServerErrorStatusCodes([500, 501, 502]);
+
+        var settings = manual.ToDictionary();
+
+        var keys = GetManualTracerSettingKeys();
+        settings.Should().ContainKeys(keys).And.HaveSameCount(keys);
+    }
+
+    [Fact]
+    public void ManualToAutomatic_CustomSettingsAreTransferredCorrectly()
+    {
+        Dictionary<string, object> initialValues = new();
+        PopulateDictionaryIntegration.PopulateSettings(initialValues, new TracerSettings());
+
+        var manual = new ManualSettings(initialValues, isFromDefaultSources: false)
+        {
+            AgentUri = new Uri("http://localhost:1234"),
+            AnalyticsEnabled = true,
+            CustomSamplingRules = """[{"sample_rate":0.3, "service":"shopping-cart.*"}]""",
+            DiagnosticSourceEnabled = true,
+            DisabledIntegrationNames = ["something"],
+            Environment = "my-test-env",
+            GlobalSamplingRate = 0.5,
+            GlobalTags = new Dictionary<string, string> { { "tag1", "value" } },
+            GrpcTags = new Dictionary<string, string> { { "grpc1", "grpc-value" } },
+            HeaderTags = new Dictionary<string, string> { { "header1", "header-value" } },
+            KafkaCreateConsumerScopeEnabled = false,
+            LogsInjectionEnabled = true,
+            MaxTracesSubmittedPerSecond = 50,
+            ServiceName = "my-test-service",
+            ServiceVersion = "1.2.3",
+            StartupDiagnosticLogEnabled = false,
+            StatsComputationEnabled = true,
+            TraceEnabled = false,
+            TracerMetricsEnabled = true,
+        };
+
+        manual.Integrations[nameof(IntegrationId.Aerospike)].Enabled = false;
+        manual.Integrations[nameof(IntegrationId.Grpc)].AnalyticsEnabled = true;
+        manual.Integrations[nameof(IntegrationId.Couchbase)].AnalyticsSampleRate = 0.5;
+
+        var mappings = new Dictionary<string, string> { { "some-service", "some-mapping" } };
+        int[] clientErrors = [400, 401, 402];
+        int[] serverErrors = [500, 501, 502];
+
+        manual.SetServiceNameMappings(mappings);
+        manual.SetHttpClientErrorStatusCodes(clientErrors);
+        manual.SetHttpServerErrorStatusCodes(serverErrors);
+
+        var changedValues = manual.ToDictionary();
+
+        var automatic = new TracerSettings();
+        ConfigureIntegration.UpdateSettings(changedValues, automatic);
+
+        AssertEquivalent(manual, automatic);
+        automatic.ServiceNameMappings.Should().Equal(mappings);
+        automatic.HttpClientErrorStatusCodes.Should().Equal(TracerSettings.ParseHttpCodesToArray(string.Join(",", clientErrors)));
+        automatic.HttpServerErrorStatusCodes.Should().Equal(TracerSettings.ParseHttpCodesToArray(string.Join(",", serverErrors)));
+    }
+
+    [Fact]
+    public void AutomaticToManual_ImmutableSettingsAreTransferredCorrectly()
```

Comment:
Can/Should we add a "round trip" that automatically compares _all_ properties (using `Should().BeEquivalentTo()`)

Reply:
Because of how the transmission happens, I'm not sure there's a great way to do that